src/main/js/dynamic/DistributionConfiguration.js

```diff
@@ -0,0 +1,315 @@
+        }
+
+        if (!nextProps.fetching && !nextProps.inProgress) {
+            if (this.loading) {
+                this.loading = false;
+                const { job } = nextProps;
+                if (job && job.fieldModels) {
+                    const channelModel = nextProps.job.fieldModels.find(model => model.descriptorName.startsWith('channel_'));
+                    const providerModel = nextProps.job.fieldModels.find(model => model.descriptorName.startsWith('provider_'));
+                    const newChannel = this.props.descriptors.find(descriptor => descriptor.name === channelModel.descriptorName && descriptor.context === DescriptorUtilities.CONTEXT_TYPE.DISTRIBUTION);
+                    const newProvider = this.props.descriptors.find(descriptor => descriptor.name === providerModel.descriptorName && descriptor.context === DescriptorUtilities.CONTEXT_TYPE.DISTRIBUTION);
+
+                    const includeAllProjects = this.flipFilterFlag(providerModel);
+                    const updatedProviderConfig = Object.assign({}, FieldModelUtilities.updateFieldModelSingleValue(providerModel, KEY_FILTER_BY_PROJECT, includeAllProjects));
+                    this.setState({
+                        fieldErrors: {},
+                        channelConfig: channelModel,
+                        providerConfig: updatedProviderConfig,
+                        currentChannel: newChannel,
+                        currentProvider: newProvider
+                    });
+                }
+            }
+        }
+    }
+
+    componentDidUpdate() {
+        const { channelConfig, currentChannel, currentProvider } = this.state;
+
+        const selectedChannelOption = FieldModelUtilities.getFieldModelSingleValue(channelConfig, KEY_CHANNEL_NAME);
+        const prevChannelName = currentChannel ? currentChannel.name : '';
+
+        if (selectedChannelOption && prevChannelName !== selectedChannelOption) {
+            const newChannel = this.props.descriptors.find(descriptor => descriptor.name === selectedChannelOption && descriptor.context === DescriptorUtilities.CONTEXT_TYPE.DISTRIBUTION);
+            const channelKeys = FieldMapping.retrieveKeys(newChannel.fields);
+            const emptyChannelConfig = FieldModelUtilities.createEmptyFieldModel(channelKeys, newChannel.context, newChannel.name);
+            const updatedChannelConfig = {};
+            Object.assign(updatedChannelConfig, FieldModelUtilities.updateFieldModelSingleValue(emptyChannelConfig, KEY_CHANNEL_NAME, selectedChannelOption));
+            const name = FieldModelUtilities.getFieldModelSingleValue(channelConfig, KEY_NAME);
+            const frequency = FieldModelUtilities.getFieldModelSingleValue(channelConfig, KEY_FREQUENCY);
+            const provider = FieldModelUtilities.getFieldModelSingleValue(channelConfig, KEY_PROVIDER_NAME);
+            Object.assign(updatedChannelConfig, FieldModelUtilities.updateFieldModelSingleValue(updatedChannelConfig, KEY_NAME, name));
```

Comment:
Do these Object.assign(...)'s need to be here? You are ignoring the object they return. I would expect something like 
const thing = Object.assign(...)

Reply:
The first argument passed is the one that gets updated. with all the other objects that are passed. So I'm updating the updatedChannelConfig object

Reply:
That may be a side effect of the Object.assign() but it is bad practice to rely on that. You should be using the returned object as the new "complete" object rather than expecting the target to be updated. Unless Object.assign specifically documents that this is alright to do.